### PR TITLE
gopkg.toml: upgrade to client-go v9.0.0 / kubernetes v1.12.6

### LINKED ISSUE #934: Upgrade to k8s.io/client-go 1.12.x
Upgrade from client-go 1.11.x to 1.12.x

### PR DESCRIPTION
**Pull request by vaamarnath**
Signed-off-by: Amarnath <vaamarnath@gmail.com>

Fixes #934

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -56,10 +56,16 @@
     "envoy/api/v2/listener",
     "envoy/api/v2/route",
     "envoy/config/accesslog/v2",
+    "envoy/config/bootstrap/v2",
```
Comment (vaamarnath) [inline Gopkg.lock]: These dependencies got added when I run `dep ensure` after updating `Gopkg.toml`. I double-checked if it is genuine by running `dep check` and looks like these were missing from `Gopkg.lock` earlier.

```
➜  contour git:(client-go-update-1.12) ✗ dep check
# Gopkg.lock is out of sync:
github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v2: imported or required, but missing from Gopkg.lock's input-imports
github.com/envoyproxy/go-control-plane/envoy/config/metrics/v2: imported or required, but missing from Gopkg.lock's input-imports
```

Comment (davecheney) [reply Gopkg.lock]: Who can say, dep is pretty buggy

**Comment by davecheney** [comment]
@vaamarnath i'll merge this today. If you want to raise a p2 issue for `dep check` complaining, please go for it. I don't think its a high priority, we'll probably move away from dep by 0.12

**Comment by vaamarnath** [comment]
@davecheney Yeah, I see there is an issue (#598) open already for moving to go modules. I will not raise an issue for `dep check` complaining for now.

**Comment by davecheney** [comment]
I’m not sure what dep check is upset about, but it’s probably been upset for a long time and nothing bad has happened. I don’t see any warnings during the Travis ci builds so I think we can live with this debt til 0.12

**Pull request merged**